### ISSUE TITLE
Log API Exceptions to Lambda Errors

### ISSUE THREAD

**Issue by mccalltd** [aws/aws-lambda-dotnet#514]
I am deploying an aspnet core web api to Lambda. All is well, but I find it annoying that unhandled server errors are not reflected in the `Errors` metric for the Lambda. Do you know of any way to do this? 

The reason this matters to me is it makes monitoring, debugging and discovery much simpler with tools that ingest CloudWatch logs and metrics (eg, Dashlane). Thanks!

**Comment by bjorg** [comment]
Application errors and Lambda errors are not semantically the same.

For example, when handling a batch of SQS messages, you must exit with an error to avoid the message from being deleted, even when some messages were successful. We manually delete successful messages to avoid them being sent again for no good reason. However, in this case, a Lambda failure just means 1 or more messages had issues, the rest could have been fine. You will need to emit custom metrics to really understand how your application works. Similarly, when processing a Kinesis stream, you cannot ever fail or you Lambda gets stuck in place. You must exit with a success and handle failures another way.

In short, application errors and Lambda errors are not semantically the same.

**Comment by ashishdhingra** [comment]
Hi @mccalltd, 

Please advise if input provided by @bjorg answers the question and if this issue could be closed.

Thanks,
Ashish
